org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/TypePattern.java

```diff
@@ -276,7 +277,11 @@ boolean subtreeMatch0(ASTMatcher matcher, Object other) {
 	ASTNode clone0(AST target) {
 		TypePattern result = new TypePattern(target);
 		result.setSourceRange(getStartPosition(), getLength());
-		result.setPatternVariable((VariableDeclaration) getPatternVariable().clone(target));
+		if (this.ast.apiLevel < AST.JLS22_INTERNAL) {
+			result.setPatternVariable((VariableDeclaration) getPatternVariable().clone(target));
```

Comment:
Look at MethodDeclaration.clone0() - see how it has internalSetModifiers() method - this avoids deprecated warnings

Reply:
Done